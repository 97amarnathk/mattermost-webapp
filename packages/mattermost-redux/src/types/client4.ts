// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

// I assume these are the log levels
export type LogLevel = 'ERROR' | 'WARNING' | 'INFO';

// TODO remove
export type ClientResponse<T> = {
    response: Response;
    headers: Map<string, string>;
    data: T;
};

type ErrorOffline = {
    message: string;
    url: string;
};
type ErrorInvalidResponse = {
    intl: {
        id: string;
        defaultMessage: string;
    };
};
export type ErrorApi = {
    message: string;
    server_error_id: string;
    status_code: number;
    url: string;
};
export type Client4Error = ErrorOffline | ErrorInvalidResponse | ErrorApi;

// TODO remove
export type Options = {
    headers?: { [x: string]: string };
    method?: string;
    url?: string;
    credentials?: 'omit' | 'same-origin' | 'include';
    body?: any;
};

// TODO remove
export type StatusOK = {
    status: 'OK';
};
