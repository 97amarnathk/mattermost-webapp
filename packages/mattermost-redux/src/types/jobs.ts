// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {IDMappedObjects} from './utilities';

// TODO remove
export type JobType = 'data_retention' | 'elasticsearch_post_indexing' | 'ldap_sync' | 'message_export';

// TODO remove
export type JobStatus = 'pending' | 'in_progress' | 'success' | 'error' | 'cancel_requested' | 'canceled';

// TODO remove
export type Job = JobTypeBase & {
    id: string;
    priority: number;
    create_at: number;
    start_at: number;
    last_activity_at: number;
    status: JobStatus;
    progress: number;
    data: any;
};

export type JobsByType = {
    [x in JobType]?: Job[];
};

export type JobsState = {
    jobs: IDMappedObjects<Job>;
    jobsByTypeList: JobsByType;
};

// TODO remove
export type JobTypeBase = {
    type: JobType;
}
