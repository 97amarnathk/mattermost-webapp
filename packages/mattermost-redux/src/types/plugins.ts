// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

// TODO remove
type Plugin = {
    id: string;
    name: string;
    description?: string;
    homepage_url?: string;
    support_url?: string;
    release_notes_url?: string;
    icon_path?: string;
    version: string;
    min_server_version?: string;
    server?: PluginManifestServer;
    backend?: PluginManifestServer;
    webapp?: PluginManifestWebapp;
    settings_schema?: PluginSettingsSchema;
    props?: Record<string, any>;
};

// TODO remove
export type PluginManifest = Plugin;

export type PluginRedux = PluginManifest & {active: boolean};

// TODO remove
export type PluginManifestServer = {
    executables?: {
        'linux-amd64'?: string;
        'darwin-amd64'?: string;
        'windows-amd64'?: string;
    };
    executable: string;
};

// TODO remove
export type PluginManifestWebapp = {
    bundle_path: string;
};

// TODO remove
export type PluginSettingsSchema = {
    header: string;
    footer: string;
    settings: PluginSetting[];
};

// TODO remove
export type PluginSetting = {
    key: string;
    display_name: string;
    type: string;
    help_text: string;
    regenerate_help_text?: string;
    placeholder: string;
    default: any;
    options?: PluginSettingOption[];
};

// TODO remove
export type PluginSettingOption = {
    display_name: string;
    value: string;
};

// TODO remove
export type PluginsResponse = {
    active: PluginManifest[];
    inactive: PluginManifest[];
};

// TODO remove
export type PluginStatus = {
    plugin_id: string;
    cluster_id: string;
    plugin_path: string;
    state: number;
    name: string;
    description: string;
    version: string;
};

type PluginInstance = {
    cluster_id: string;
    version: string;
    state: number;
}

export type PluginStatusRedux = {
    id: string;
    name: string;
    description: string;
    version: string;
    active: boolean;
    state: number;
    instances: PluginInstance[];
}

// TODO remove
export type ClientPluginManifest = {
    id: string;
    min_server_version?: string;
    version: string;
    webapp: {
        bundle_path: string;
    };
}

export type MarketplaceLabel = { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    name: string;
    description?: string;
    url?: string;
    color?: string;
}

export enum HostingType { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    OnPrem = 'on-prem',
    Cloud = 'cloud',
}

export enum AuthorType { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    Mattermost = 'mattermost',
    Partner = 'partner',
    Community = 'community',
}

export enum ReleaseStage { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    Production = 'production',
    Beta = 'beta',
    Experimental = 'experimental',
}

export type MarketplacePlugin = { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    homepage_url?: string;
    icon_data?: string;
    download_url?: string;
    release_notes_url?: string;
    labels?: MarketplaceLabel[];
    hosting?: HostingType;
    author_type: AuthorType;
    release_stage: ReleaseStage;
    enterprise: boolean;
    manifest: PluginManifest;
    installed_version?: string;
}
