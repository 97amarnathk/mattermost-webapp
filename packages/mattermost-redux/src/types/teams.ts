// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {ServerError} from './errors';
import {UserProfile} from './users';
import {Dictionary, RelationOneToOne} from './utilities';

// TODO remove
export type TeamMembership = {
    mention_count: number;
    msg_count: number;
    mention_count_root: number;
    msg_count_root: number;
    team_id: string;
    user_id: string;
    roles: string;
    delete_at: number;
    scheme_user: boolean;
    scheme_admin: boolean;
};

// TODO remove
export type TeamMemberWithError = {
    member: TeamMembership;
    user_id: string;
    error: ServerError;
}

// TODO remove
export type TeamType = 'O' | 'I';

// TODO remove
export type Team = {
    id: string;
    create_at: number;
    update_at: number;
    delete_at: number;
    display_name: string;
    name: string;
    description: string;
    email: string;
    type: TeamType;
    company_name: string;
    allowed_domains: string;
    invite_id: string;
    allow_open_invite: boolean;
    scheme_id: string;
    group_constrained: boolean;
    policy_id?: string | null;
};

export type TeamsState = {
    currentTeamId: string;
    teams: Dictionary<Team>;
    myMembers: Dictionary<TeamMembership>;
    membersInTeam: RelationOneToOne<Team, RelationOneToOne<UserProfile, TeamMembership>>;
    stats: RelationOneToOne<Team, TeamStats>;
    groupsAssociatedToTeam: any;
    totalCount: number;
};

// TODO
export type TeamUnread = {
    team_id: string;
    mention_count: number;
    msg_count: number;
    mention_count_root: number;
    msg_count_root: number;
};

// TODO
export type GetTeamMembersOpts = {
    sort?: 'Username';
    exclude_deleted_users?: boolean;
};

// TODO
export type TeamsWithCount = {
    teams: Team[];
    total_count: number;
};

// TODO
export type TeamStats = {
    team_id: string;
    total_member_count: number;
    active_member_count: number;
};

// TODO remove
export type TeamSearchOpts = {
    page?: number;
    per_page?: number;
    allow_open_invite?: boolean;
    group_constrained?: boolean;
}

// TODO remove
export type TeamInviteWithError = {
    email: string;
    error: ServerError;
};
