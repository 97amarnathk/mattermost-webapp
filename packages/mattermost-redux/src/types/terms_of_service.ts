// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

// TODO rmeove all

export type TermsOfService = {
    id: string;
    create_at: number;
    user_id: string;
    text: string;
}
