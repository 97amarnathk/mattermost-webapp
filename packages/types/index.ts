// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export * from './src/admin';
export * from './src/apps';
export * from './src/audit';
export * from './src/autocomplete';
export * from './src/bots';
export * from './src/channel_categories';
export * from './src/channels';
export * from './src/cloud';
export * from './src/compliance';
export * from './src/config';
export * from './src/data_retention';
export * from './src/emojis';
export * from './src/errors';
export * from './src/files';
export * from './src/general';
export * from './src/groups';
export * from './src/integration_actions';
export * from './src/integrations';
export * from './src/jobs';
export * from './src/marketplace';
export * from './src/message_attachments';
export * from './src/mfa';
export * from './src/plugins';
export * from './src/posts';
export * from './src/preferences';
export * from './src/product_notices';
export * from './src/reactions';
export * from './src/roles';
export * from './src/saml';
export * from './src/schemes';
export * from './src/sessions';
export * from './src/teams';
export * from './src/terms_of_service';
export * from './src/threads';
export * from './src/users';
