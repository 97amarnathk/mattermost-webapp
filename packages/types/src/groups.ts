// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {UserProfile} from './users';

export type SyncablePatch = {
    scheme_admin: boolean;
    auto_add: boolean;
};

export type GroupPatch = {
    allow_reference: boolean;
    name: string;
};

export type Group = {
    id: string;
    name: string;
    display_name: string;
    description: string;
    type: string;
    remote_id: string;
    create_at: number;
    update_at: number;
    delete_at: number;
    has_syncables: boolean;
    member_count: number;
    scheme_admin: boolean;
    allow_reference: boolean;
};

export type GroupSyncable = {
    group_id: string;

    auto_add: boolean;
    scheme_admin: boolean;
    create_at: number;
    delete_at: number;
    update_at: number;
    type: 'Team' | 'Channel';
};

export type MixedUnlinkedGroup = {
    mattermost_group_id?: string;
    name: string;
    primary_key: string;
    has_syncables?: boolean;
};

export type UserWithGroup = UserProfile & {
    groups: Group[];
    scheme_guest: boolean;
    scheme_user: boolean;
    scheme_admin: boolean;
};

export type GroupsWithCount = {
    groups: Group[];
    total_group_count: number;

    // These fields are added by the client after the groups are returned by the server
    channelID?: string;
    teamID?: string;
}

export type UsersWithGroupsAndCount = {
    users: UserWithGroup[];
    total_count: number;
};
