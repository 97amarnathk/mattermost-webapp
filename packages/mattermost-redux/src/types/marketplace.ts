// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

// This file's contents belong to the Apps Framework feature.
// Apps Framework feature is experimental, and the contents of this file are
// susceptible to breaking changes without pushing the major version of this package.
import {PluginManifest} from './plugins';
import {AppManifest} from './apps';

// TODO remove all

// TODO remove
export type MarketplaceLabel = {
    name: string;
    description?: string;
    url?: string;
    color?: string;
}

// TODO remove
export enum HostingType {
    OnPrem = 'on-prem',
    Cloud = 'cloud',
}

// TODO remove
export enum AuthorType {
    Mattermost = 'mattermost',
    Partner = 'partner',
    Community = 'community',
}

// TODO remove
export enum ReleaseStage {
    Production = 'production',
    Beta = 'beta',
    Experimental = 'experimental',
}

// TODO remove
interface MarketplaceBaseItem {
    labels?: MarketplaceLabel[];
    hosting?: HostingType;
    author_type: AuthorType;
    release_stage: ReleaseStage;
    enterprise: boolean;
}

export interface MarketplacePlugin extends MarketplaceBaseItem {
    manifest: PluginManifest;
    icon_data?: string;
    homepage_url?: string;
    download_url?: string;
    release_notes_url?: string;
    installed_version?: string;
}

// TODO remove
export interface MarketplaceApp extends MarketplaceBaseItem {
    manifest: AppManifest;
    installed: boolean;
}
