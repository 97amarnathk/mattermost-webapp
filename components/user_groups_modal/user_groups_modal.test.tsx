// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';

import {shallow} from 'enzyme';

import {Group, GroupPermissions} from 'mattermost-redux/types/groups';

import UserGroupsModal from './user_groups_modal';

describe('component/user_groups_modal', () => {
    const baseProps = {
        onExited: jest.fn(),
        groups: [],
        myGroups: [],
        searchTerm: '',
        currentUserId: '',
        backButtonAction: jest.fn(),
        groupPermissionsMap: {},
        canCreateCustomGroups: true,
        actions: {
            openModal: jest.fn(),
            getGroups: jest.fn(),
            setModalSearchTerm: jest.fn(),
            getGroupsByUserIdPaginated: jest.fn(),
            searchGroups: jest.fn(),
            addUsersToGroup: jest.fn(),
            removeUsersFromGroup: jest.fn(),
            archiveGroup: jest.fn(),
        },
    };

    function getGroups(numberOfGroups: number) {
        const groups: Group[] = [];
        for (let i = 0; i < numberOfGroups; i++) {
            groups.push({
                id: `group${i}`,
                name: `group${i}`,
                display_name: `Group ${i}`,
                description: `Group ${i} description`,
                source: 'custom',
                remote_id: null,
                create_at: 1637349374137,
                update_at: 1637349374137,
                delete_at: 0,
                has_syncables: false,
                member_count: i + 1,
                allow_reference: true,
                scheme_admin: false,
            });
        }

        return groups;
    }

    function getPermissions(groups: Group[], myGroups: Group[]) {
        const groupPermissionsMap: Record<string, GroupPermissions> = {};
        [...groups, ...myGroups].forEach((g) => {
            groupPermissionsMap[g.id] = {
                can_delete: true,
                can_manage_members: true,
            };
        });

        return groupPermissionsMap;
    }

    test('should match snapshot without groups', () => {
        const wrapper = shallow(
            <UserGroupsModal
                {...baseProps}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with groups', () => {
        const groups = getGroups(3);
        const myGroups = getGroups(1);
        const permissions = getPermissions(groups, myGroups);

        const wrapper = shallow(
            <UserGroupsModal
                {...baseProps}
                groups={groups}
                myGroups={myGroups}
                groupPermissionsMap={permissions}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with groups, myGroups selected', () => {
        const groups = getGroups(3);
        const myGroups = getGroups(1);
        const permissions = getPermissions(groups, myGroups);

        const wrapper = shallow(
            <UserGroupsModal
                {...baseProps}
                groups={getGroups(3)}
                myGroups={getGroups(1)}
                groupPermissionsMap={permissions}
            />,
        );

        wrapper.setState({selectedFilter: 'my'});

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with groups, search group1', () => {
        const groups = getGroups(3);
        const myGroups = getGroups(1);
        const permissions = getPermissions(groups, myGroups);

        const wrapper = shallow(
            <UserGroupsModal
                {...baseProps}
                groups={getGroups(3)}
                myGroups={getGroups(1)}
                groupPermissionsMap={permissions}
                searchTerm='group1'
            />,
        );

        const instance = wrapper.instance() as UserGroupsModal;

        const e = {
            target: {
                value: '',
            },
        };
        instance.handleSearch(e as React.ChangeEvent<HTMLInputElement>);
        expect(baseProps.actions.setModalSearchTerm).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.setModalSearchTerm).toBeCalledWith('');

        e.target.value = 'group1';
        instance.handleSearch(e as React.ChangeEvent<HTMLInputElement>);
        expect(wrapper.state('loading')).toEqual(true);
        expect(baseProps.actions.setModalSearchTerm).toHaveBeenCalledTimes(2);
        expect(baseProps.actions.setModalSearchTerm).toBeCalledWith(e.target.value);
    });
});
