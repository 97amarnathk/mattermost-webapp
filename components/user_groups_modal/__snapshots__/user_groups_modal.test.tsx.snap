// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/user_groups_modal should match snapshot with groups 1`] = `
<Modal
  animation={true}
  aria-labelledby="userGroupsModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal user-groups-modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  id="userGroupsModal"
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[MockFunction]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="dialog"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="userGroupsModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="User Groups"
        id="user_groups_modal.title"
      />
    </ModalTitle>
    <a
      className="user-groups-create btn btn-md btn-primary"
      href="#"
      id="test"
      onClick={[Function]}
    >
      <MemoizedFormattedMessage
        defaultMessage="Create Group"
        id="user_groups_modal.createNew"
      />
    </a>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="user-groups-search"
    >
      <div
        className="user-groups-searchbar"
      >
        <span
          aria-hidden="true"
          className="user-groups-search-icon"
        >
          <SearchIcon />
        </span>
        <input
          data-testid="searchInput"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <i
          className="user-groups-clear-button fa fa-times-circle hidden"
          data-testid="clear-search"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="more-modal__dropdown"
    >
      <MenuWrapper
        animationComponent={[Function]}
        className=""
        id="groupsFilterDropdown"
      >
        <a>
          <span>
            Show: All Groups
          </span>
          <span
            className="caret"
          />
        </a>
        <Menu
          ariaLabel="Groups Filter Menu"
          openLeft={false}
        >
          <MenuItemAction
            buttonClass="groups-filter-btn"
            id="groupsDropdownAll"
            onClick={[Function]}
            rightDecorator={
              <i
                className="icon icon-check"
              />
            }
            show={true}
            text="All Groups"
          />
          <MenuItemAction
            buttonClass="groups-filter-btn"
            id="groupsDropdownMy"
            onClick={[Function]}
            rightDecorator={false}
            show={true}
            text="My Groups"
          />
        </Menu>
      </MenuWrapper>
    </div>
    <div
      className="user-groups-modal__content user-groups-list"
      onScroll={[Function]}
    >
      <div
        className="group-row"
        key="group0"
        onClick={[Function]}
      >
        <span
          className="group-display-name"
        >
          Group 0
        </span>
        <span
          className="group-name"
        >
          @
          group0
        </span>
        <div
          className="group-member-count"
        >
          <MemoizedFormattedMessage
            defaultMessage="{member_count} {member_count, plural, one {member} other {members}}"
            id="user_groups_modal.memberCount"
            values={
              Object {
                "member_count": 1,
              }
            }
          />
        </div>
        <div
          className="group-action"
        >
          <MenuWrapper
            animationComponent={[Function]}
            className=""
            id="customWrapper-group0"
            isDisabled={false}
            stopPropagationOnToggle={true}
          >
            <div
              className="text-right"
            >
              <button
                className="action-wrapper"
              >
                <i
                  className="icon icon-dots-vertical"
                />
              </button>
            </div>
            <Menu
              ariaLabel="User Actions Menu"
              className="group-actions-menu"
              openLeft={true}
              openUp={false}
            >
              <MenuGroup>
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-account-multiple-outline"
                    />
                  }
                  onClick={[Function]}
                  show={true}
                  text="View Group"
                />
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-account-multiple-outline"
                    />
                  }
                  onClick={[Function]}
                  show={true}
                  text="Join Group"
                />
              </MenuGroup>
              <MenuGroup>
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-exit-to-app"
                    />
                  }
                  isDangerous={true}
                  onClick={[Function]}
                  show={true}
                  text="Leave Group"
                />
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-archive-outline"
                    />
                  }
                  isDangerous={true}
                  onClick={[Function]}
                  show={true}
                  text="Archive Group"
                />
              </MenuGroup>
            </Menu>
          </MenuWrapper>
        </div>
      </div>
      <div
        className="group-row"
        key="group1"
        onClick={[Function]}
      >
        <span
          className="group-display-name"
        >
          Group 1
        </span>
        <span
          className="group-name"
        >
          @
          group1
        </span>
        <div
          className="group-member-count"
        >
          <MemoizedFormattedMessage
            defaultMessage="{member_count} {member_count, plural, one {member} other {members}}"
            id="user_groups_modal.memberCount"
            values={
              Object {
                "member_count": 2,
              }
            }
          />
        </div>
        <div
          className="group-action"
        >
          <MenuWrapper
            animationComponent={[Function]}
            className=""
            id="customWrapper-group1"
            isDisabled={false}
            stopPropagationOnToggle={true}
          >
            <div
              className="text-right"
            >
              <button
                className="action-wrapper"
              >
                <i
                  className="icon icon-dots-vertical"
                />
              </button>
            </div>
            <Menu
              ariaLabel="User Actions Menu"
              className="group-actions-menu"
              openLeft={true}
              openUp={false}
            >
              <MenuGroup>
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-account-multiple-outline"
                    />
                  }
                  onClick={[Function]}
                  show={true}
                  text="View Group"
                />
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-account-multiple-outline"
                    />
                  }
                  onClick={[Function]}
                  show={true}
                  text="Join Group"
                />
              </MenuGroup>
              <MenuGroup>
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-exit-to-app"
                    />
                  }
                  isDangerous={true}
                  onClick={[Function]}
                  show={true}
                  text="Leave Group"
                />
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-archive-outline"
                    />
                  }
                  isDangerous={true}
                  onClick={[Function]}
                  show={true}
                  text="Archive Group"
                />
              </MenuGroup>
            </Menu>
          </MenuWrapper>
        </div>
      </div>
      <div
        className="group-row"
        key="group2"
        onClick={[Function]}
      >
        <span
          className="group-display-name"
        >
          Group 2
        </span>
        <span
          className="group-name"
        >
          @
          group2
        </span>
        <div
          className="group-member-count"
        >
          <MemoizedFormattedMessage
            defaultMessage="{member_count} {member_count, plural, one {member} other {members}}"
            id="user_groups_modal.memberCount"
            values={
              Object {
                "member_count": 3,
              }
            }
          />
        </div>
        <div
          className="group-action"
        >
          <MenuWrapper
            animationComponent={[Function]}
            className=""
            id="customWrapper-group2"
            isDisabled={false}
            stopPropagationOnToggle={true}
          >
            <div
              className="text-right"
            >
              <button
                className="action-wrapper"
              >
                <i
                  className="icon icon-dots-vertical"
                />
              </button>
            </div>
            <Menu
              ariaLabel="User Actions Menu"
              className="group-actions-menu"
              openLeft={true}
              openUp={false}
            >
              <MenuGroup>
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-account-multiple-outline"
                    />
                  }
                  onClick={[Function]}
                  show={true}
                  text="View Group"
                />
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-account-multiple-outline"
                    />
                  }
                  onClick={[Function]}
                  show={true}
                  text="Join Group"
                />
              </MenuGroup>
              <MenuGroup>
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-exit-to-app"
                    />
                  }
                  isDangerous={true}
                  onClick={[Function]}
                  show={true}
                  text="Leave Group"
                />
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-archive-outline"
                    />
                  }
                  isDangerous={true}
                  onClick={[Function]}
                  show={true}
                  text="Archive Group"
                />
              </MenuGroup>
            </Menu>
          </MenuWrapper>
        </div>
      </div>
      <LoadingScreen
        position="relative"
        style={Object {}}
      />
    </div>
  </ModalBody>
</Modal>
`;

exports[`component/user_groups_modal should match snapshot with groups, myGroups selected 1`] = `
<Modal
  animation={true}
  aria-labelledby="userGroupsModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal user-groups-modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  id="userGroupsModal"
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[MockFunction]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="dialog"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="userGroupsModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="User Groups"
        id="user_groups_modal.title"
      />
    </ModalTitle>
    <a
      className="user-groups-create btn btn-md btn-primary"
      href="#"
      id="test"
      onClick={[Function]}
    >
      <MemoizedFormattedMessage
        defaultMessage="Create Group"
        id="user_groups_modal.createNew"
      />
    </a>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="user-groups-search"
    >
      <div
        className="user-groups-searchbar"
      >
        <span
          aria-hidden="true"
          className="user-groups-search-icon"
        >
          <SearchIcon />
        </span>
        <input
          data-testid="searchInput"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <i
          className="user-groups-clear-button fa fa-times-circle hidden"
          data-testid="clear-search"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="more-modal__dropdown"
    >
      <MenuWrapper
        animationComponent={[Function]}
        className=""
        id="groupsFilterDropdown"
      >
        <a>
          <span>
            Show: My Groups
          </span>
          <span
            className="caret"
          />
        </a>
        <Menu
          ariaLabel="Groups Filter Menu"
          openLeft={false}
        >
          <MenuItemAction
            buttonClass="groups-filter-btn"
            id="groupsDropdownAll"
            onClick={[Function]}
            rightDecorator={false}
            show={true}
            text="All Groups"
          />
          <MenuItemAction
            buttonClass="groups-filter-btn"
            id="groupsDropdownMy"
            onClick={[Function]}
            rightDecorator={
              <i
                className="icon icon-check"
              />
            }
            show={true}
            text="My Groups"
          />
        </Menu>
      </MenuWrapper>
    </div>
    <div
      className="user-groups-modal__content user-groups-list"
      onScroll={[Function]}
    >
      <div
        className="group-row"
        key="group0"
        onClick={[Function]}
      >
        <span
          className="group-display-name"
        >
          Group 0
        </span>
        <span
          className="group-name"
        >
          @
          group0
        </span>
        <div
          className="group-member-count"
        >
          <MemoizedFormattedMessage
            defaultMessage="{member_count} {member_count, plural, one {member} other {members}}"
            id="user_groups_modal.memberCount"
            values={
              Object {
                "member_count": 1,
              }
            }
          />
        </div>
        <div
          className="group-action"
        >
          <MenuWrapper
            animationComponent={[Function]}
            className=""
            id="customWrapper-group0"
            isDisabled={false}
            stopPropagationOnToggle={true}
          >
            <div
              className="text-right"
            >
              <button
                className="action-wrapper"
              >
                <i
                  className="icon icon-dots-vertical"
                />
              </button>
            </div>
            <Menu
              ariaLabel="User Actions Menu"
              className="group-actions-menu"
              openLeft={true}
              openUp={false}
            >
              <MenuGroup>
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-account-multiple-outline"
                    />
                  }
                  onClick={[Function]}
                  show={true}
                  text="View Group"
                />
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-account-multiple-outline"
                    />
                  }
                  onClick={[Function]}
                  show={true}
                  text="Join Group"
                />
              </MenuGroup>
              <MenuGroup>
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-exit-to-app"
                    />
                  }
                  isDangerous={true}
                  onClick={[Function]}
                  show={true}
                  text="Leave Group"
                />
                <MenuItemAction
                  disabled={false}
                  icon={
                    <i
                      className="icon-archive-outline"
                    />
                  }
                  isDangerous={true}
                  onClick={[Function]}
                  show={true}
                  text="Archive Group"
                />
              </MenuGroup>
            </Menu>
          </MenuWrapper>
        </div>
      </div>
      <LoadingScreen
        position="relative"
        style={Object {}}
      />
    </div>
  </ModalBody>
</Modal>
`;

exports[`component/user_groups_modal should match snapshot without groups 1`] = `
<Modal
  animation={true}
  aria-labelledby="userGroupsModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal user-groups-modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  id="userGroupsModal"
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[MockFunction]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="dialog"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="userGroupsModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="User Groups"
        id="user_groups_modal.title"
      />
    </ModalTitle>
    <button
      className="user-groups-create btn btn-md btn-primary"
      onClick={[Function]}
    >
      <MemoizedFormattedMessage
        defaultMessage="Create Group"
        id="user_groups_modal.createNew"
      />
    </button>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="user-groups-search"
    >
      <SearchIcon />
      <Input
        className="user-group-search-input"
        data-testid="searchInput"
        onChange={[Function]}
        placeholder="Search Groups"
        type="text"
        value=""
      />
    </div>
    <div
      className="more-modal__dropdown"
    >
      <MenuWrapper
        animationComponent={[Function]}
        className=""
        id="groupsFilterDropdown"
      >
        <a>
          <span>
            Show: All Groups
          </span>
          <span
            className="icon icon-chevron-down"
          />
        </a>
        <Menu
          ariaLabel="Groups Filter Menu"
          openLeft={false}
        >
          <MenuItemAction
            buttonClass="groups-filter-btn"
            id="groupsDropdownAll"
            onClick={[Function]}
            rightDecorator={
              <i
                className="icon icon-check"
              />
            }
            show={true}
            text="All Groups"
          />
          <MenuItemAction
            buttonClass="groups-filter-btn"
            id="groupsDropdownMy"
            onClick={[Function]}
            rightDecorator={false}
            show={true}
            text="My Groups"
          />
        </Menu>
      </MenuWrapper>
    </div>
    <div
      className="user-groups-modal__content user-groups-list"
      onScroll={[Function]}
    >
      <LoadingScreen
        position="relative"
        style={Object {}}
      />
    </div>
  </ModalBody>
</Modal>
`;
